rfcs/0000-User&HostIntegrationRFC.md

```diff
@@ -0,0 +1,248 @@
+Feel free to remove these comments as you go along.
+-->
+
+<!--
+Stage 0: Provide a high level summary of the premise of these changes. Briefly describe the nature, purpose, and impact of the changes. ~2-5 sentences.
+-->
+
+This proposal aims to extend the existing ECS fieldset to store inventory metadata for hosts, and users from external application repositories. Using ECS to store such fields will improve metadata querying and retrieval across various use cases.
+
+Within the Security `Entity Analytics` initiative, we refer to hosts and users as `entities`. We refer to hosts/ users as `assets` across generic security or observability use cases. For simplification, entities, and assets will refer to host and user objects.
+
+This proposal includes the following:
+* Additional fields in the `users` object 
+* Introduces a new field set called `assets`
+<!-- * Additional fields in the `host` object --->
+
+This proposal will also facilitate storing of host and user inventory within the security solution (aka. the entity store).
+
+
+<!--
+Stage 1: If the changes include field additions or modifications, please create a folder titled as the RFC number under rfcs/text/. This will be where proposed schema changes as standalone YAML files or extended example mappings and larger source documents will go as the RFC is iterated upon.
+-->
+
+<!--
+Stage X: Provide a brief explanation of why the proposal is being marked as abandoned. This is useful context for anyone revisiting this proposal or considering similar changes later on.
+-->
+
+## Fields
+
+<!--
+Stage 1: Describe at a high level how this change affects fields. Include new or updated yml field definitions for all of the essential fields in this draft. While not exhaustive, the fields documented here should be comprehensive enough to deeply evaluate the technical considerations of this change. The goal here is to validate the technical details for all essential fields and to provide a basis for adding experimental field definitions to the schema. Use GitHub code blocks with yml syntax formatting, and add them to the corresponding RFC folder.
+-->
+
+### Proposed New Fields for User object
+
+Field | Type | Example | Description
+--- | --- | --- | ---
+user.profile.id	| keyword	| 1234	| User ID from the identity datasource.
+user.profile.type	| keyword	| Employee	| Type of user account.
+user.profile.status	| keyword |	On board	| Status of the user account.
+user.profile.first_name	| keyword |	First	| First Name of the User.
+user.profile.last_name	| keyword |	Last	| Last Name of the user.
+user.profile.other_identities	| keyword , multifield |	first.last@elk.elastic.co	| Array of additional user identities.
+user.profile.manager	| keyword |	John Doe	| Assigned Manager for the user account.
+user.profile.employee_type	| keyword |	Regular | Further classification type for the user account.
```

Comment:
Capturing this level of user profile data as part of an observability system is extremely problematic due to the PII nature of this data.

The `user.profile.*` listed here is more like the information that would be stored (and protected) by the identity system and only available on the backend and as part of the encrypted details in any token.

i.e. They should not be included with general observability data.

Reply:
I would then recommend that a note (at the least) be added to these fields. Identifying there is possible PII issues that sending / storing this data might cause.

e.g. sending / storing this may lead to EUDB violations for example (depending on lots of caveats and how the data is store / whether it can be removed (right to forget) etc.).

 For some people / use cases this might be an acceptable option, but we don't really want to have anyone just including the data because they can.

Reply:
👋 @MSNev, thanks for taking the time to review. Great observations.

This subset of the user.* may be most prudent for security use cases and likely not in general observability use cases. Depending on the identity application mappings and definitions, these fields may also contain PII data.

I proposed these to users.* schema changes for three primary reasons:
1. This security use case of ingesting Identity metadata assumes appropriate index-level access within ElasticSearch and by the security solution.
2. ECS does not restrict whether a field should/should not be populated with PII. 
3. Theoretically, if you allow me to generalize, the data mapping definitions are abstracted from how the data is consumed within an application (viewed, queried, etc.)

We have previously discussed general approaches for pseudonymizing data within the ECS framework. Thus, I recommend standardizing these mapping definitions and expanding the data anonymization concepts in parallel. 

Please lmk if these thoughts resonate with you or if further clarifications are needed.